From e4ee47faca97429b7cf9de32ed8af2295b2befec Mon Sep 17 00:00:00 2001
From: Philip-Dylan Gleonec <philip-dylan.gleonec@savoirfairelinux.com>
Date: Fri, 2 Apr 2021 11:23:37 +0200
Subject: [PATCH] services: move conf db out of /etc

Move conf.db out of /etc/openvswitch to /var/lib/openvswitch.
This enables to have /etc as a read-only mountpoint for ovs-vswitchd
and ovsdb-server services.
---
 rhel/usr_lib_systemd_system_ovs-vswitchd.service.in | 1 +
 rhel/usr_lib_systemd_system_ovsdb-server.service    | 1 +
 2 files changed, 2 insertions(+)

diff --git a/rhel/usr_lib_systemd_system_ovs-vswitchd.service.in b/rhel/usr_lib_systemd_system_ovs-vswitchd.service.in
index f88b57cae..8cfe5f2bf 100644
--- a/rhel/usr_lib_systemd_system_ovs-vswitchd.service.in
+++ b/rhel/usr_lib_systemd_system_ovs-vswitchd.service.in
@@ -11,6 +11,7 @@ PartOf=openvswitch.service
 Type=forking
 PIDFile=/var/run/openvswitch/ovs-vswitchd.pid
 Restart=on-failure
+Environment="OVS_USER_OPT=--db-file=/var/lib/openvswitch/conf.db"
 Environment=XDG_RUNTIME_DIR=/var/run/openvswitch
 EnvironmentFile=/etc/openvswitch/default.conf
 EnvironmentFile=-/etc/sysconfig/openvswitch
diff --git a/rhel/usr_lib_systemd_system_ovsdb-server.service b/rhel/usr_lib_systemd_system_ovsdb-server.service
index 69b0bf61a..d83506bcf 100644
--- a/rhel/usr_lib_systemd_system_ovsdb-server.service
+++ b/rhel/usr_lib_systemd_system_ovsdb-server.service
@@ -9,6 +9,7 @@ PartOf=openvswitch.service
 Type=forking
 PIDFile=/var/run/openvswitch/ovsdb-server.pid
 Restart=on-failure
+Environment="OVS_USER_OPT=--db-file=/var/lib/openvswitch/conf.db"
 EnvironmentFile=/etc/openvswitch/default.conf
 EnvironmentFile=-/etc/sysconfig/openvswitch
 ExecStartPre=-/bin/mkdir /var/run/openvswitch /var/log/openvswitch
-- 
2.25.1

